urllib3/util/ssl_.py

```diff
@@ -17,6 +18,26 @@
 
 import errno
 import warnings
+import hmac
+
+if hasattr(hmac, 'compare_digest'):
+    _const_compare_digest = hmac.compare_digest
+else:
+    def _const_compare_digest(a, b):
+        """
+        Compare two digests of equal length in constant time.
+
+        Returns True if the digests match, and False otherwise.
+        """
+        if len(a) != len(b):
+            raise ValueError("Mismatched input lengths!")
+
+        result = 0
+        for l, r in six.moves.zip(bytearray(a), bytearray(b)):
```

Comment:
Nah, it'll just truncate to the shortest value. Same thing that `six.moves.zip` will do. In py2, it's just not a generator, which I don't think we care about.